### ISSUE TITLE
autonight doesn't work when changing settings

### ISSUE THREAD

**Issue by danielk117** [OpenIPC/device-mjsxj02hl#2]
the autonight script doesn't seems to work right, when changing a setting and majestic restarts. it stays in day mode, but the cam was in a dark room.

**Comment by kasitoru** [comment]
Try it now

**Comment by danielk117** [comment]
No, it stays in day mode.

**Comment by kasitoru** [comment]
Execute in:
```
/etc/init.d/S96autonight stop
/usr/sbin/autonight.sh
```
And without closing the console, perform the actions that lead to the problem. Post the output here.

**Comment by kasitoru** [comment]
Also test this https://t.me/mjsxj02hl_firmware/5706

**Comment by danielk117** [comment]
> Execute in:
> 
> ```
> /etc/init.d/S96autonight stop
> /usr/sbin/autonight.sh
> ```
> 
> And without closing the console, perform the actions that lead to the problem. Post the output here.

Stopped service. Running script directly. Change a setting in majestic using web UI. Nothings happens, no output on console.

```
~# /usr/sbin/autonight.sh
....................
Watching at isp_again > 14000 to enable night mode
Watching at isp_again < 2000 to disable night mode
Polling interval: 5 sec
....................
```

> Also test this https://t.me/mjsxj02hl_firmware/5706

Following your instruction, IR light and filter stays on, when majestic was killed. When restarting majestic, both is reseted.

**Comment by kasitoru** [comment]
> Following your instruction, IR light and filter stays on, when majestic was killed. When restarting majestic, both is reseted.

Picture is grayscale after restarting majestic? What is the value `night_enabled` in output `http://<cam_ip>/metrics`

**Comment by danielk117** [comment]
No, pictures is colored after restarting majestic. Value of `night_enabled` is `0`.

**Comment by kasitoru** [comment]
Very strange. My camera has a completely different behavior. Let's compare actions:

1. `/etc/init.d/S96autonight stop`
2. Manually turn on night mode (Web interface -> Preview -> Push "Toggle night mode" button)
3. `killall -1 majestic` (argument -1 is important)
4. View picture color and `night_enabled` value (http://<cam_ip>/metrics)

**Comment by danielk117** [comment]
> 3\. `killall -1 majestic` (argument -1 is important)

you're right. i was using `killall majestic` and restarted it manually. discribed your way, I'am getting a colored picture but `night_enabled` value is `1`.

**Comment by kasitoru** [comment]
This is probably a bug in majestic https://github.com/OpenIPC/majestic/issues/84
